### DISCUSSION TITLE
Could not locate Comsol installation

### DISCUSSION THREAD

**Discussion by isaacbasil** [MPh-py/MPh#138] [General]
Hi, 

I am having some trouble locating the comsol installation when calling mph.start()... the error I get is "RuntimeError: Could not locate any Comsol installation."

It seems strange, as comsol is installed on the same machine that I am running the Python script on. Indeed, it isn't installed in the default location, however, I have added some lines to my bashrc file to help mph find the shared libraries (shown below). Additionally, entering the command 'comsol' in my terminal successfully starts up the application, and I am using version 1.2.3 of the MPh package, which I believe includes comsol installations found on PATH in discovery. 

Any advice would be appreciated :)

Best,
Isaac

lines in bashrc: 
```
# Help MPh find Comsol's shared libraries.
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:\
/opt/COMSOL/6.0/multiphysics/lib/glnxa64:\
/opt/COMSOL/6.0/multiphysics/ext/graphicsmagick/glnxa64:\
/opt/COMSOL/6.0/multiphysics/ext/impi/glnxa64:\
/opt/COMSOL/6.0/multiphysics/ext/sqlite/glnxa64
```

Specs: 
Linux UBUNTU 18.04.6 LTS (GNU/Linux 5.4.0-96-generic x86_64)
COMSOL 6.0

**Comment by john-hen** [comment]
Hi. First off, no need to add those extra lines to `.bashrc`. This would only help when trying to run the Comsol client in "stand-alone" mode on Linux/macOS. Which doesn't really work, or only barely, even if Comsol were found. Which is why it's only ever mentioned in chapter ["Limitations"](https://mph.readthedocs.io/en/1.2/limitations.html). You can remove those lines. For the record, they don't help MPh find "Comsol", but rather Comsol's "shared libraries". That's different, for reasons that are not important in this context.

You are right in that the discovery "is expected" to work in your case. I haven't tested on Ubuntu 18 with Comsol 6.0, but don't see how that would make a difference.

MPh comes with a [test suite](https://github.com/MPh-py/MPh/tree/main/tests). If you download (or `git clone`) the source code, you can test the discovery with log output enabled, to hopefully gain some insight into what's going wrong. This is what it looks like on Linux CentOS 7, where Comsol is (also) installed in a custom location, the `comsol` command is (also) on the search path:
```console
❯ python tools/test.py discovery --log
Running test group "discovery".
[00:00.174] Searching system for available Comsol back-ends.
[00:00.174] Searching file system for Comsol executables.
[00:00.174] Looking for Comsol executable on system search path.
[00:00.174] Running shell command "which comsol".
[00:00.178] Command response was "/cluster/apps/nss/comsol/6.0/x86_64/bin/comsol".
[00:00.178] Found Comsol executable "/cluster/apps/nss/comsol/6.0/x86_64/bin/comsol".
[00:00.178] Checking executable "/cluster/apps/nss/comsol/6.0/x86_64/bin/comsol".
[00:00.179] Found Java bridge configuration "/cluster/apps/nss/comsol/6.0/x86_64/bin/glnxa64/comsol.ini".
[00:00.179] System architecture is "glnxa64".
[00:00.180] Java VM at relative path "../../java/glnxa64/jre/lib/server/libjvm.so".
[00:00.181] Root folder is "/cluster/apps/nss/comsol/6.0/x86_64".
[00:00.963] Reported version information is "COMSOL Multiphysics 6.0.0.318".
[00:00.963] Assigned name "6.0" to this installation.
Passed in 1 s.
```

**Reply by john-hen** [reply]
PS: To format code as such in comments here, use Markdown "code fences", that is triple backticks. For example:
````
```
code goes here
```
````

I've edited your original comment (which I usually would refrain from doing) to add those backticks where missing, just to make it more readable. (Since a comment in bash, starting with `#`, marks a heading in Markdown.)

**Reply by isaacbasil** [reply]
Thanks so much for the quick response :) That's very helpful - I'll go give it a try. And regarding the comment formatting, thanks for letting me know also. 
Isaac
